synapse/config/server.py

```diff
@@ -979,6 +979,11 @@ def generate_config_section(
           # (e.g. every 12h), but not want that purge to be performed by a job that's
           # iterating over every room it knows, which could be heavy on the server.
           #
+          # If any purge job is configured, it is strongly recommended to have at least
+          # a single job with neither 'shortest_max_lifetime' nor 'longest_max_lifetime'
+          # set, or one job without 'shortest_max_lifetime' and one job without
+          # 'longest_max_lifetime' set.
```

Comment:
The reason I've added this comment is because, with this change, it is possible that the retention policy that's stored in the DB for a room is outside of the bounds allowed by the config, since the database reflects the values sent in the `m.room.retention` event and that value gets corrected when starting the purges. I don't really know if this should be explained in the configuration comment, though, and if so how it should be phrased.

FTR the reason for this is because I don't think a policy stored in the DB should depend on the config that was in place when it was set, as it would mean the corrected `max_lifetime` might not be determined using the current config and that we don't have a standard formula at a given time to correct every room's policies. This would also mean a mistake in the config would be permanently stored in the DB and not easily reverted.

Reply:
I think that approach makes sense (storing the event data and modifying it at purge time). I'll think a bit more about whether the config comments should be updated.

Reply:
I think it is worth explaining in this comment why this is necessary (or what might happen if you *don't* do this).

I find the way these purge jobs are defined to be really confusing, but that's not a result of this PR.